Sources/_StringProcessing/RegexDSL/Core.swift

```diff
@@ -167,10 +145,8 @@ extension RegexProtocol {
         return nil
       }
       let convertedMatch: Match
-      if Match.self == Tuple2<Substring, DynamicCaptures>.self {
-        convertedMatch = Tuple2(
-          input[range], DynamicCaptures(captures)
-        ) as! Match
+      if Match.self == (Substring, DynamicCaptures).self {
+        convertedMatch = (input[range], DynamicCaptures(captures)) as! Match
```

Comment:
I was wondering, what is this code path for?

Reply:
See `testDynamicCaptures()`. This is used for compiling and running a regex string that's not known at compile time (https://github.com/apple/swift-experimental-string-processing/pull/36).